CMakeLists.txt

```diff
@@ -61,6 +61,28 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
   endif()
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+  # Determine if the host is running an illumos distribution:
+  execute_process(COMMAND /usr/bin/uname -o OUTPUT_VARIABLE UNAME_O
```

Comment:
A few nits:
* Is `uname` always located at `/usr/bin/uname` on a "SunOS" host?
* Re: cross-compiling -- the target system could be "SunOS", but the build host system might be something different. Should we condition this on `CMAKE_HOST_SYSTEM_NAME STREQUAL "SunOS"` also being true? 
* Re: cross-compiling -- is there a use-case for the build host being "illumos" but the target is a different "SunOS" variant?

I don't consider these concerns to be blocking.

Reply:
@jclulow is going to answer these questions (he has far more historical context than I) but I'm pushing a commit to address (2), since it doesn't make a lot of sense to try and detect illumos via uname if you're running on Linux. Good catch!

Reply:
* Yes, `/bin` has been a link to `/usr/bin` on illumos (and our ancestral Solaris) for at least 20 years.  It's important to get the system-provided `uname` here because for reasons that I will never understand sometimes people will ship the GNU `uname` as well, in a directory that ends up in `PATH` first on their system, and the output of `guname -o` is ... unreliable at best!  (It has often said **Solaris** on illumos systems, which is pretty much the opposite of helpful)
* Yes, your point about cross-compiling is well-taken!  What we really want is to be a first class citizen in CMake at this point, I think, so that we can have the framework itself work these things out without executing additional commands on the build host or confusing cross compilation, etc.  I have filed CMake issue [26302 first class support for illumos systems](https://gitlab.kitware.com/cmake/cmake/-/issues/26302) to try and get the ball rolling there, but I expect it will take quite some time to get that to the point where we can just rely on it instead.

Thanks for looking at all this, we really appreciate it!